src/languageFacts/colors.ts

```diff
@@ -683,59 +732,146 @@ export function XYZtoLAB(xyz: XYZ, round: Boolean = true): LAB {
 	const lab: LAB = XYZtoLAB(xyz, round);
 	return lab;
 }
-export function lchFromColor(rgba: Color): LCH {
-	const lab: LAB = labFromColor(rgba, false);
+
+export function oklabFromColor(rgba: Color, round = true): LAB {
+	const xyz: XYZ = RGBtoXYZ(rgba);
+	const lab: LAB = XYZtoOKLAB(xyz, round);
+	// Convert lightness to a percentage of oklab
+	return { ...lab, l: lab.l * 100 };
+}
+
+/**
+ * Calculate chroma and hue from Lab values
+ * Returns LCH values without formatting/rounding
+ */
+function labToLCH(lab: LAB): LCH {
 	const c: number = Math.sqrt(Math.pow(lab.a, 2) + Math.pow(lab.b, 2));
-	let h: number = Math.atan2(lab.b, lab.a) * (180 / Math.PI);
+	let h: number = Math.atan2(lab.b, lab.a) * RADIANS_TO_DEGREES_FACTOR;
 	while (h < 0) {
 		h = h + 360;
 	}
 	return {
-		l: Math.round((lab.l + Number.EPSILON) * 100) / 100,
-		c: Math.round((c + Number.EPSILON) * 100) / 100,
-		h: Math.round((h + Number.EPSILON) * 100) / 100,
-		alpha: lab.alpha
+		l: lab.l,
+		c: c,
+		h: h,
+		alpha: lab.alpha,
 	};
 }
 
-export function colorFromLAB(l: number, a: number, b: number, alpha: number = 1.0): Color {
-	const lab: LAB = {
-		l,
-		a,
-		b,
-		alpha
+export function lchFromColor(rgba: Color): LCH {
+	const lab: LAB = labFromColor(rgba, false);
+	const lch: LCH = labToLCH(lab);
+
+	return {
+		l: Math.round((lch.l + Number.EPSILON) * 100) / 100,
+		c: Math.round((lch.c + Number.EPSILON) * 100) / 100,
+		h: Math.round((lch.h + Number.EPSILON) * 100) / 100,
+		alpha: lch.alpha,
+	};
+}
+
+export function oklchFromColor(rgba: Color): LCH {
+	const lab: LAB = oklabFromColor(rgba, false);
+	const lch: LCH = labToLCH(lab);
+
+	return {
+		l: Number((lch.l).toFixed(3)),
+		c: Number(lch.c.toFixed(5)),
+		h: Number(lch.h.toFixed(3)),
+		alpha: lch.alpha,
 	};
-	const xyz = xyzFromLAB(lab);
+}
+
+/**
+ * Generic function to convert LAB/OKLAB to Color
+ */
+function labToColor(lab: LAB, xyzConverter: (lab: LAB) => XYZ): Color {
+	const xyz = xyzConverter(lab);
 	const rgb = xyzToRGB(xyz);
 	return {
-		red: (rgb.red >= 0 ? (rgb.red <= 255 ? rgb.red : 255) : 0) / 255.0,
-		green: (rgb.green >= 0 ? (rgb.green <= 255 ? rgb.green : 255) : 0) / 255.0,
-		blue: (rgb.blue >= 0 ? (rgb.blue <= 255 ? rgb.blue : 255) : 0) / 255.0,
-		alpha
+		red: (rgb.red >= 0 ? Math.min(rgb.red, 255) : 0) / 255,
+		green: (rgb.green >= 0 ? Math.min(rgb.green, 255) : 0) / 255,
+		blue: (rgb.blue >= 0 ? Math.min(rgb.blue, 255) : 0) / 255,
+		alpha: lab.alpha ?? 1,
 	};
 }
 
+export function colorFromLAB(l: number, a: number, b: number, alpha = 1): Color {
+	return labToColor({ l, a, b, alpha }, xyzFromLAB);
+}
+
+export function colorFromOKLAB(l: number, a: number, b: number, alpha = 1): Color {
+	return labToColor({ l, a, b, alpha }, xyzFromOKLAB);
+}
+
 export interface LAB { l: number; a: number; b: number; alpha?: number; }
 
-export function labFromLCH(l: number, c: number, h: number, alpha: number = 1.0): LAB {
+const DEGREES_TO_RADIANS_FACTOR = Math.PI / 180;
```

Comment:
```suggestion
const DEGREES_TO_RADIANS_FACTOR = Math.PI / (DEGREES_PER_CIRCLE / 2);
```

I think there should be a minor change here to re-use the constants better. Would you like that added to this PR, or a new one?

Reply:
Could also be `const DEGREES_TO_RADIANS_FACTOR = Math.PI / DEGREES_PER_CIRCLE * 2;`

If fewer parenthesis are preferred.

Reply:
You can make a follow-up PR, if you want. I think the brackets help, even if not strictly necessary.